Check LSDF Mount, VM, and Website

kalebphipps opened the pull request:
# Description

This PR includes a script that automatically checks if the LSDF mount, the VM, and the paint-database.org website are working.

The script is executed via ``crontab`` every 10 minutes and sends E-Mails if there is a failure.

## Type of change

<!-- Please delete options that are not relevant. -->

- [X] New feature (non-breaking change which adds functionality)


# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

Markus-Goetz commented on html/check_website_and_mount.py:
```diff
@@ -0,0 +1,130 @@
+# Configuration
+MOUNT_POINT = "/mnt/lsdf"
+WEBSITE_URL = "https://paint-database.org"
+VM_HOST = "141.52.215.85"
```
Do we have a static IP or can this change?

kalebphipps replied:
I was given a static IP - I think that was a requirement to enable the NFS mount (even though that is not set up yet). So I am assuming we have a static IP - would have to check with Nico though on that one!

Markus-Goetz commented on html/check_website_and_mount.py:
```diff
@@ -0,0 +1,130 @@
+LOCK_FILE = "/tmp/monitor.lock"
+
+# Initialize email content
+email_subject = "🚨 PAINT ALERT - There is a Problem with the PAINT Database 🚨"
```
I would make this a constant variable, i.e. capital letters

kalebphipps replied:
It's not a constant and not needed - therefore removed

Markus-Goetz commented on html/check_website_and_mount.py:
```diff
@@ -0,0 +1,130 @@
+
+# Initialize email content
+email_subject = "🚨 PAINT ALERT - There is a Problem with the PAINT Database 🚨"
+email_body = []
```
This is not a constant variable and should rather be moved to main and appended to/passed into the functions

kalebphipps replied:
Changed

Markus-Goetz commented on html/check_website_and_mount.py:
```diff
@@ -0,0 +1,130 @@
+    # Check if the lock file exists.
+    if os.path.exists(LOCK_FILE):
+        # If it exists, exit without performing any checks.
+        sys.exit(0)
```
Should this output a warning that the lock file should be removed?

kalebphipps replied:
Not quite - the lock file shouldn't be removed if the error has not been fixed. I have changed the logic now so that if all tests are passing and the lock file is still there we now get a warning!

Markus-Goetz commented on html/check_website_and_mount.py:
```diff
@@ -0,0 +1,130 @@
+
+        sys.exit(1)  # Exit with an error status to indicate a problem.
+
+    # Silent exit if no errors.
```
This happens automatically not necessary

kalebphipps replied:
Removed

Markus-Goetz commented on html/check_website_and_mount.py:
```diff
@@ -0,0 +1,130 @@
+        sys.exit(1)  # Exit with an error status to indicate a problem.
+
+    # Silent exit if no errors.
+    sys.exit(0)
```
Newline at the end of the file

kalebphipps replied:
The newline is automatically there (or should be there, at least it is in pycharm) due to the Ruff formatting...

kalebphipps commented:
I have cleaned up the code and added in an additional warning if everything appears to be working but the lock file is still present.

Waiting on confirmation that the IPs are static...
